src/Microsoft.DotNet.Build.Tasks.Feed/PushToBlobFeed.cs

```diff
@@ -53,22 +53,26 @@ public override bool Execute()
 
                     if (!SkipCreateContainer)
                     {
-                        await blobFeedAction.CreateContainerAsync(this.BuildEngine);
+                        await blobFeedAction.CreateContainerAsync(BuildEngine, PublishFlatContainer);
                     }
 
-                    List<string> items = ConvertToStringLists(ItemsToPush);
-
-                    if (!PublishFlatContainer)
+                    if (PublishFlatContainer)
                     {
-                        await blobFeedAction.PushToFeed(items, Overwrite);
+                        await PublishToFlatContainerAsync(ItemsToPush, blobFeedAction);
                     }
                     else
                     {
-                        using (var clientThrottle = new SemaphoreSlim(this.MaxClients, this.MaxClients))
+                        ITaskItem[] symbolItems = ItemsToPush.Where(i => i.ItemSpec.Contains("symbols.nupkg")).Select(i => 
                         {
-                            Log.LogMessage($"Uploading {ItemsToPush.Length} items...");
-                            await Task.WhenAll(ItemsToPush.Select(item => blobFeedAction.UploadAssets(item, clientThrottle, Overwrite)));
-                        }
+                            string fileName = Path.GetFileName(i.ItemSpec);
+                            i.SetMetadata("RelativeBlobPath", $"symbols/{fileName}");
+                            return i;
+                        }).ToArray();
```

Comment:
From an efficiency point of view you are calling ToArray and then passing an IEnumerable and later calling ToList you are making a number of copies of these lists. That may start to add up for larger file lists.